### PR TITLE
Parentheses always appear in pairs.

### PR DESCRIPTION
**Pull request by frihed**
Note: Other mods may also have this problem.

### CODE REVIEW

File: src/column.rs [hunk 1]
```diff
@@ -15,6 +16,7 @@ pub enum FunctionExpression {
     Max(FunctionArguments),
     Min(FunctionArguments),
     GroupConcat(FunctionArguments, String),
+    Concat(Vec<FieldDefinitionExpression>),
```
Comment (ms705) [inline src/column.rs:19]: The use of `FieldDefinitionExpression` here seems potentially troublesome. `FieldDefinitionExpression` encompasses variants `All` (which is equivalent to `*` in a select query) and `AllInTable` (which is `table.*`), but neither of these work with MySQL's `concat()` function:
```
mysql> select concat (email, roles) from ContactInfo;
+-----------------------+
| concat (email, roles) |
+-----------------------+
| malte@cs.brown.edu5   |
+-----------------------+
1 row in set (0.01 sec)

mysql> select concat (ContactInfo) from ContactInfo;
ERROR 1054 (42S22): Unknown column 'ContactInfo' in 'field list'
mysql> select concat (ContactInfo.*) from ContactInfo;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '*) from ContactInfo' at line 1
mysql> select concat (*) from ContactInfo;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '*) from ContactInfo' at line 1
```

In other words, we should only allow columns and literals here. Perhaps we need a variant of `FieldDefinitionExpression` that only supports column names and literals.

File: src/common.rs [hunk 2]
```diff
@@ -972,8 +976,10 @@ pub fn literal(i: &[u8]) -> IResult<&[u8], Literal> {
 
 pub fn literal_expression(i: &[u8]) -> IResult<&[u8], LiteralExpression> {
     map(
-        pair(
-            delimited(opt(tag("(")), literal, opt(tag(")"))),
+        pair(alt((
+                delimited(tag("("), literal, tag(")")),
+                literal
+                )),
```
Comment (ms705) [inline src/common.rs:982]: Thanks for catching this!

File: src/common.rs [hunk 3]
```diff
@@ -1091,6 +1097,21 @@ mod tests {
...
+        let res = column_function(qs);
+        let expected = FunctionExpression::Concat(vec![
+            FieldDefinitionExpression::Col(Column::from("first_name")),
+            FieldDefinitionExpression::Value(FieldValueExpression::Literal(LiteralExpression::from(Literal::String(" ".to_string())))),
+            FieldDefinitionExpression::Col(Column::from("last_name")),
+        ]);
+        assert_eq!(res.unwrap().1, expected);
+        assert_eq!(format!("{}", column_function(qs1.as_bytes()).unwrap().1), qs1);
+    }
+
```
Comment (ms705) [inline src/common.rs:1114]: Might be good to add an expected-to-fail test for an incorrectly bracketed variant here. Check that `res.is_err()` is true.

Comment (frihed) [reply src/common.rs:1114]: Thank you for your instruction and advice. I've learned a lot.

**Pull request closed without merging**